src/core/control/tools/EraseHandler.cpp

```diff
@@ -131,7 +134,8 @@ void EraseHandler::eraseStroke(Layer* l, Stroke* s, double x, double y, Range& r
             return;
         }
         const double paddingCoeff = PADDING_COEFFICIENT_CAP[s->getStrokeCapStyle()];
-        const PaddedBox paddedEraserBox{{x, y}, halfEraserSize, halfEraserSize + paddingCoeff * s->getWidth()};
+        const double widthMultiplier = s->hasPressure() ? settings->getPressureMultiplier() : 1.;
```

Comment:
Thanks @rolandlo! I incorporated your patch (and tagged you as a coauthor), except I modified this line.

Reply:
I should point out that this is only a fix that works in most cases. It doesn't quite work if modifying the pressure multiplier between drawing and erasing strokes and it doesn't quite work if the stylus reports pressure values above 1.0. 
A cleaner fix would be to compute/cache the maximal pressure value among all points of the stroke and use it as a `widthMultiplier`. That has some performance cost though.

So should we go with the "easy fix" or incorporate the cleaner fix?

Reply:
> So should we go with the "easy fix" or incorporate the cleaner fix?

I'm sure we can come up with a better fix than this. The information of what stroke segments have just been erased is available somewhere. We just have to use that instead of rerendering a dumb square.
I'll give this a try this weekend.